src/components/general/mediaTypes/ImageViewer.tsx

```diff
@@ -0,0 +1,319 @@
+                /* Return to cursor tool */
+                dispatch(setAnnotoriousMode('cursor'));
+
+                /* Set selected annotation */
+                setEditAnnotation(w3cAnnotation);
+
+                /* Focus on input field */
+                setTimeout(() => {
+                    tooltipFieldRef.current?.focus();
+                }, 300);
+            });
+
+            /* Event for when an Annotatio has been edited */
+            anno.on('updateAnnotation', (W3cAnnotation: ImageAnnotation) => {
+                /* Set new edit target */
+                setEditAnnotation(W3cAnnotation);
+            });
+        }
+    }, [anno, image]);
+
+    /* Function for refreshing annotations on Annotorious layer */
+    const RefreshAnnotations = () => {
+        /* If there are visual annotations present, calculate positions and redraw */
+        if (!isEmpty(digitalMediaAnnotations.visual)) {
+            const visualImageAnnotations = CalculateAnnotationPositions();
+
+            anno.setAnnotations(visualImageAnnotations);
+        } else {
+            anno.setAnnotations([]);
+        }
+    }
+
+    useEffect(() => {
+        if (anno) {
+            RefreshAnnotations();
+        }
+    }, [digitalMediaAnnotations]);
+
+    /* Function for depicting method when selecting an Annotation */
+    const SelectAction = (w3cAnnotation: ImageAnnotation) => {
+        let selectAction;
+
+        /* Enable Annotorious edit mode if logged in and Annotation belongs to user, otherwise highlight */
+        if (KeycloakService.IsLoggedIn() && (KeycloakService.GetSubject() === w3cAnnotation.bodies[0].creator.id)) {
+            selectAction = PointerSelectAction.EDIT;
+        } else {
+            selectAction = PointerSelectAction.HIGHLIGHT;
+        }
+
+        return selectAction;
+    }
+
+    /* Function for calculating annotation positions */
+    const CalculateAnnotationPositions = () => {
+        const visualImageAnnotations: Dict[] = [];
+
+        if (image) {
+            /* Get original size of image */
+            const imgWidth = image.width;
+            const imgHeight = image.height;
+
+            /* For each annotation, calculate the W3C pixels relative to the TDWG AC position */
+            digitalMediaAnnotations.visual.forEach((imageAnnotation: Annotation) => {
+                /* Check if the annotation is a visual one */
+                if (imageAnnotation.target.selector?.hasROI) {
+                    const ROI = imageAnnotation.target.selector.hasROI;
+
+                    const x = ROI["ac:xFrac"] * imgWidth;
+                    const y = ROI["ac:yFrac"] * imgHeight;
+                    const w = ROI["ac:widthFrac"] * imgWidth;
+                    const h = ROI["ac:heightFrac"] * imgHeight;
+
+                    visualImageAnnotations.push({
+                        id: imageAnnotation.id,
+                        "@context": 'http://www.w3.org/ns/anno.jsonld',
```

Comment:
Not sure about this context field

Reply:
Also to feed Annotorious, we might be able to use another context, I took this over from the author's code